app/controllers/announcements_controller.rb

```diff
@@ -0,0 +1,46 @@
+class AnnouncementsController < ApplicationController
+  before_action :find_message, only: [:show, :edit, :update, :destroy]
+
+  def index
+    @announcements = Announcement.all.order("created_at DESC")
+  end
+
+  def show
+    @announcement = Announcement.find(params[:id])
```

Comment:
isn't this the same code as in find_message, and you call a before_action on find_message